gammapy/background/energy_offset_array.py

```diff
@@ -104,3 +127,76 @@ def plot_image(self, ax=None, offset=None, energy=None, **kwargs):
         image = ax.imshow(self.data, extent=extent, **kwargs)
         plt.colorbar(image)
         return ax
+
+    @classmethod
+    def read(cls, filename):
+        """Create `EnergyOffsetArray` from  fits file.
+
+        Parameters
+        ----------
+        filename : str
+            File name
+        """
+        hdu_list = fits.open(filename)
+        return cls.from_fits(hdu_list)
+
+    
+    @classmethod
+    def from_fits(cls, hdu, scheme=None):
+        """Read EnergyOffsetArray from a fits binary table.
+
+        Parameters
+        ----------
+        hdu : `~astropy.io.fits.BinTableHDU`
+            HDU binary table for the EnergyOffsetArray.
+        Returns
+        -------
+        cube : `~gammapy.background.EnergyOffsetArray`
+            EnergyOffsetArray object.
+        """
+
+        header = hdu[1].header
+        data = hdu[1].data
+        
+        # get offset and energy binning
+        offset_edges = _make_bin_edges_array(data['THETA_LO'], data['THETA_HI'])
+        energy_edges = _make_bin_edges_array(data['ENERG_LO'], data['ENERG_HI'])
+        
+        # get data
+        energy_offset_array=data['EnergyOffsetArray'][0]
+        return cls(offset_edges, energy_edges , energy_offset_array)
+
+    def to_table(self):
+        """Convert EnergyOffsetArray to astropy table format.
+
+        Returns
+        -------
+        table : `~astropy.table.Table`
+            Table containing the EnergyOffsetArray.
+        """
+        # table
+        table = Table()
+        table['THETA_LO']=np.expand_dims(self.offset[:-1],0)
```

Comment:
I think the way it works here
https://github.com/gammapy/gammapy/blob/master/gammapy/background/cube.py#L383
is that the extra length-1 dimension at the start that's needed to store it in a BINTABLE single row is added by putting the array in a list with one entry (that's then converted to an array of length 1.

Example:
```
>>> import numpy as np
>>> offset = np.array([1,2,3])
>>> offset
array([1, 2, 3])
>>> offset.shape
(3,)
>>> [offset]
[array([1, 2, 3])]
>>> np.array([offset])
array([[1, 2, 3]])
>>> np.array([offset]).shape
(1, 3)
```
Can you try using the same code here?
Example:
```python
table['THETA_LO'] = [self.offset[:-1]]
```